### PR TITLE
chore: ignore close callbacks for clicked

### PR DESCRIPTION
**Pull request by Shahroz16**
changes:

- Before sending click event to journeys ignore actions that are for close.

### CODE REVIEW

File: messaginginapp/src/test/java/io/customer/messaginginapp/ModuleMessagingInAppTest.kt [hunk 1]
```diff
@@ -43,14 +45,15 @@ internal class ModuleMessagingInAppTest : JUnitTest() {
                 diGraph {
                     sdk {
                         overrideDependency<ScopeProvider>(testScopeProviderStub)
+                        val spykEventBus = spyk(eventBus)
+                        overrideDependency<EventBus>(spykEventBus)
```
Comment (mrehan27) [inline messaginginapp/src/test/java/io/customer/messaginginapp/ModuleMessagingInAppTest.kt:49]: Could be simplified?

```suggestion
                        overrideDependency<EventBus>(spyk(eventBus))
```

File: messaginginapp/src/test/java/io/customer/messaginginapp/ModuleMessagingInAppTest.kt [hunk 2]
```diff
@@ -184,4 +187,54 @@ internal class ModuleMessagingInAppTest : JUnitTest() {
...
+                    deliveryID = "test_campaign_id",
+                    event = Metric.Clicked,
+                    params = mapOf("action_name" to "Test Action", "action_value" to "test_action")
+                )
+            )
+        }
+    }
+
+    @Test
+    fun onAction_givenCloseAction_expectNoTrackInAppMetricEventPublished() = runTest {
```
Comment (mrehan27) [inline messaginginapp/src/test/java/io/customer/messaginginapp/ModuleMessagingInAppTest.kt:219]: Can we add `ExperimentalCoroutinesApi` annotation to fix the warning?

```
This declaration needs opt-in. Its usage should be marked with '@kotlinx.coroutines.ExperimentalCoroutinesApi' or '@OptIn(kotlinx.coroutines.ExperimentalCoroutinesApi::class)'
```

**Review by mrehan27 — APPROVED** [review]
Looks good

File: messaginginapp/src/test/java/io/customer/messaginginapp/ModuleMessagingInAppTest.kt (hunk 2 above)
Comment (Shahroz16) [reply messaginginapp/src/test/java/io/customer/messaginginapp/ModuleMessagingInAppTest.kt:220]: can do

**Pull request merged**